### PR TITLE
Update sample

### PR DESCRIPTION
**Pull request by MangelMaxime**
@alfonsogarciacaro I am trying to make this sample a little more helpful. Like a basic UI and some interaction system as for now it's just display the default sample of Fable.Template. And some people was looking for a more "complete" sample.

But I encounter a problem, I can't build `Main` project.

I got an error telling me: 

```
C:\Users\mange\Workspace\Github\fable-compiler\fable-electron\src\Main\Main.fsproj : error MSB4057: Target "_Inspect_FscArgs" do not exist in the project.
```

I found a reference to this problem here: https://github.com/ionide/ionide-vscode-fsharp/issues/427

But I don't understand why `Main` project fail to build when `Renderer` succeed.

### CODE REVIEW

**Comment by alfonsogarciacaro** [comment]
This is really weird. I think it's a Paket issue (pinging @matthid, @forki), because I also get the following error when running `dotnet build` (even though `dotnet restore` run without problems):

```/Users/alfonsogarciacaronunez/dev/samples-electron/src/Main/Main.fsproj : error MSB4057: The target "Restore" does not exist in the project.```

I've tried many things: changing the target framework to netstandard2.0 and netstandard1.6, renaming the project and the folder, fresh clone of the project, reverting Paket version (btw, after doing this it's really difficult to go back to latest Paket release using the magic mode as there's a cache somewhere that will stubbornly try to use the beta version)... but the only thing thad made `dotnet build` work was removing Paket and using Nuget to add the dependencies :confused:

I find also really confusing that the Renderer project with more or less the same project structure and dependencies build without problems. I was wondering if this had to do with the folder name change from `main` to `Main`, but renaming the folder or running `git clean -xfd` (or even a fresh clone) in case there were cached files somewhere didn't help 😞

**Comment by forki** [comment]
I saw this in my own projects as well. Usually something to do with netcore
installation ad it runs on some PCs.

/cc @enricosada

Am 03.09.2017 12:45 nachm. schrieb "Alfonso Garcia-Caro" <
notifications@github.com>:

This is really weird. I think it's a Paket issue (pinging @matthid
<https://github.com/matthid>, @forki <https://github.com/forki>), because I
also get the following error when running dotnet build (even though dotnet
restore run without problems):

/Users/alfonsogarciacaronunez/dev/samples-electron/src/Main/Main.fsproj :
error MSB4057: The target "Restore" does not exist in the project.

I've tried many things: changing the target framework to netstandard2.0 and
netstandard1.6, renaming the project and the folder, fresh clone of the
project, reverting Paket version (btw, after doing this it's really
difficult to go back to latest Paket release using the magic mode as
there's a cache somewhere that will stubbornly try to use the beta
version)... but the only thing thad made dotnet build work was removing
Paket and using Nuget to add the dependencies 😕

I find also really confusing that the Renderer project with more or less
the same project structure and dependencies build without problems. I was
wondering if this had to do with the folder name change from main to Main,
but renaming the folder or running git clean -xfd (or even a fresh clone)
in case there were cached files somewhere didn't help 😞

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/fable-compiler/samples-electron/pull/23#issuecomment-326797408>,
or mute the thread

**Comment by matthid** [comment]
Is there a sln file somewhere? Did you try recreating it with latest sdk2 tooling?

**Comment by matthid** [comment]
Also is there a FSharp.SDK reference anywhere? If yes you need to get rid of it completely and everywhere to properly work with SDK2

File: paket.lock [hunk 1]
```diff
@@ -1,112 +1,171 @@
...
-      Microsoft.NETCore.DotNetHostPolicy (>= 2.0)
-      Microsoft.NETCore.Platforms (>= 2.0)
-      NETStandard.Library (>= 2.0)
-    Microsoft.NETCore.DotNetAppHost (2.0)
-    Microsoft.NETCore.DotNetHostPolicy (2.0)
+      Fable.Core (>= 1.1.10) - restriction: >= netstandard1.6
+      Fable.Import.Browser (>= 0.1) - restriction: >= netstandard1.6
+      Fable.Import.Node (>= 0.1.2) - restriction: >= netstandard1.6
+      FSharp.Core (>= 4.2.1) - restriction: >= netstandard1.6
+      FSharp.NET.Sdk (>= 1.0.5) - restriction: >= netstandard1.6
```
Comment (matthid) [inline paket.lock]: This needs to go

File: paket.lock [hunk 2]
```diff
@@ -1,112 +1,171 @@
...
+      System.Runtime.Extensions (>= 4.1) - restriction: >= netstandard1.6
+      System.Runtime.Numerics (>= 4.0.1) - restriction: >= netstandard1.6
+      System.Text.RegularExpressions (>= 4.1) - restriction: >= netstandard1.6
+      System.Threading (>= 4.0.11) - restriction: >= netstandard1.6
+      System.Threading.Tasks (>= 4.0.11) - restriction: >= netstandard1.6
+      System.Threading.Tasks.Parallel (>= 4.0.1) - restriction: >= netstandard1.6
+      System.Threading.Thread (>= 4.0) - restriction: >= netstandard1.6
+      System.Threading.ThreadPool (>= 4.0.10) - restriction: >= netstandard1.6
+      System.Threading.Timer (>= 4.0.1) - restriction: >= netstandard1.6
+    FSharp.NET.Sdk (1.0.5) - restriction: >= netstandard1.6
```
Comment (matthid) [inline paket.lock]: Same

**Comment by matthid** [comment]
I think `Fable.Import.Electron` needs to be updated to remove the `FSharp.NET.Sdk` dependency, before this can work properly

**Comment by matthid** [comment]
If you have a bootstrapping problem I think you can edit the lockfile by hand...

**Comment by matthid** [comment]
> This is really weird. I think it's a Paket issue

Actually it is the result of a long outstanding issue: https://github.com/fsprojects/Paket/issues/2370
First it helped because the world was moving to `FSharp.NET.Sdk` and that additional reference made stuff "work"

Not it hurts as most are moving away from it /shrug

**Comment by alfonsogarciacaro** [comment]
Thanks for the explanation, @matthid! Now I'm trying to push a new version of Fable.Import.Electron, but apparently @MangelMaxime is the only owner of the package ;)

**Comment by MangelMaxime** [comment]
@matthid Thanks for the explanations.

@alfonsogarciacaro I am not at home so can't push a new version of Fable.Import.Electron but I re-send you an invitation.

**Comment by alfonsogarciacaro** [comment]
Thanks @MangelMaxime, I sent PR #24  to your PR :wink: This seems to be working now :tada: Thanks @matthid and @forki for your help ❤️ (now we know we need to republish Fable libraries) and thanks again Maxime for updating the sample 💪

**Comment by alfonsogarciacaro** [comment]
@matthid Sorry, just one question more. I'm getting this a lot with the magic mode. Is there any way to fix it?

```
~/d/fable-compiler.github.io ❯❯❯ mono .paket/paket.exe install
/Users/alfonsogarciacaronunez/.local/share/NuGet/Cache/Paket/5.92.3/paket.exe already exists (Github - cached(temporarily ignore updates))
```

The only solutions I've found so far are:

- Delete `/Users/alfonsogarciacaronunez/.local/share/NuGet/Cache/Paket`, but the message appears again soon after
- Make the Paket version explicit in paket.dependencies
- Disable magic mode by downloading the paket.exe from a specific release

**Comment by matthid** [comment]
I rarely see messages from the bootstrapper. Maybe we should move this particular message into the verbose mode as it doesn't seem to indicate any invalid situation. @vbfox is our expert, but maybe I broke that :)

**Comment by alfonsogarciacaro** [comment]
Sorry, I should have been clearer: I'm getting that **and only that** :wink: Meaning that Paket won't run after the message is displayed.

**Comment by MangelMaxime** [comment]
Thanks you @alfonsogarciacaro I will take a look at the PR this evening.

**Comment by vbfox** [comment]
I need to track this. I think it was @matthid that did an analysis in an issue that concluded that it's due to parallel execution of the bootstrapper.

It sound right as before magic mode it was pretty rare for the bootstrapper to run in parallel.

And the fact that it happens "sometimes" hint to that: The problem occurs on the first build after a new paket version was released, and multiple bootstrapper instances trying to update to it at the same time.

I never see it as I always pin paket version (kinda necessary for repeatable builds, after all that's why we have a lock file) but few people seem to do that :) I'll try to reproduce & fix it.

**Comment by MangelMaxime** [comment]
@alfonsogarciacaro I think I encounter a new bug.

```fs
    ERROR in ../Renderer/Renderer.fs
    C:/Users/mange/Workspace/Github/fable-compiler/fable-electron/src/Renderer/Renderer.fs(57,22): (57,82) error FABLE: Cannot find replacement for Fable.PowerPack.Date.Format::format
     @ ../Renderer/Renderer.fsproj 1:0-30
```

I don't know if I mess the Fable.Powperck release or if it's a Fable 1.2.x bug. If you could take a look would be nice :) I checked my `.nuget/packages` folder and I see the right version Fable.Powerpack here.

Perhaps it have something to do with the `internal` instructions I used on [this file](https://github.com/fable-compiler/fable-powerpack/blob/master/src/Date/Format.fs#L9-L44).

**Comment by MangelMaxime** [comment]
@alfonsogarciacaro This is ready to be merged. If you can test the demo on your mac would be nice :) I have only tested the application under windows.

I remove Fable.Powerpack as it was not needed in the end so the issue with it doesn't exist anymore in this repo.

**Pull request merged**